service/src/main/java/eu/europa/ec/fisheries/ers/service/facatch/evolution/CatchProgressHandler.java

```diff
@@ -0,0 +1,52 @@
+package eu.europa.ec.fisheries.ers.service.facatch.evolution;
+
+import eu.europa.ec.fisheries.ers.fa.entities.FaCatchEntity;
+import eu.europa.ec.fisheries.ers.fa.entities.FishingActivityEntity;
+import eu.europa.ec.fisheries.ers.service.dto.fishingtrip.catchprogress.CatchProgressDTO;
+import eu.europa.ec.fisheries.ers.service.dto.fishingtrip.catchprogress.DetailedCatchProgressDTO;
+import eu.europa.ec.fisheries.ers.service.dto.fishingtrip.catchprogress.SimpleCatchProgressDTO;
+import eu.europa.ec.fisheries.uvms.activity.model.schemas.FaCatchTypeEnum;
+import org.apache.commons.lang3.EnumUtils;
+
+import java.util.Set;
+
+
+public abstract class CatchProgressHandler {
+
+    public abstract CatchProgressDTO prepareCatchProgressDTO(FishingActivityEntity fishingActivity);
+
+    protected void handleSimpleCatch(FaCatchEntity faCatch, SimpleCatchProgressDTO catchProgressDTO) {
+        catchProgressDTO.updateTotal(faCatch.getSpeciesCode(), faCatch.getCalculatedWeightMeasure());
+    }
+
+    protected void handleSubtractingCatch(FaCatchEntity faCatch, Set<FaCatchEntity> faCatches, SimpleCatchProgressDTO catchProgressDTO) {
+        FaCatchTypeEnum faCatchType = EnumUtils.getEnum(FaCatchTypeEnum.class, faCatch.getTypeCode());
+        if(faCatchType == FaCatchTypeEnum.ONBOARD || faCatchType == FaCatchTypeEnum.KEPT_IN_NET || faCatchType == FaCatchTypeEnum.BY_CATCH || faCatchType == FaCatchTypeEnum.LOADED){
```

Comment:
for readability and maintainability reasons I would suggest that if/if else statements be replaced with methods encapsulating business logic ex isCatchLoadedAndOnboard(){ return booleanExpression;}